src/main/java/codeu/model/data/NemoBot.java

```diff
@@ -22,7 +23,7 @@
   /**
   * Constructs a new Bot.
   */
-  public NemoBot(String filePath) {
+  public NemoBot() {
```

Comment:
You're removing the file from the constructor here, but passing it up in the listener.